ext/openssl/openssl.stub.php

```diff
@@ -14,77 +14,74 @@ final class OpenSSLAsymmetricKey
 
-/** @param OpenSSLAsymmetricKey $out */
-function openssl_csr_export(OpenSSLCertificateSigningRequest|string $csr, &$out, bool $notext = true): bool {}
+/** @param OpenSSLAsymmetricKey $output */
+function openssl_csr_export(OpenSSLCertificateSigningRequest|string $request, &$output, bool $no_description = true): bool {}
 
-/** @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $priv_key */
-function openssl_csr_sign(OpenSSLCertificateSigningRequest|string $csr, OpenSSLCertificate|string|null $cacert, $priv_key, int $days, ?array $config_args = null, int $serial = 0): OpenSSLCertificate|false {}
+/** @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $private_key */
+function openssl_csr_sign(OpenSSLCertificateSigningRequest|string $request, OpenSSLCertificate|string|null $ca_certificate, $private_key, int $days, ?array $options = null, int $serial = 0): OpenSSLCertificate|false {}
 
-/** @param OpenSSLAsymmetricKey $privkey */
-function openssl_csr_new(array $dn, &$privkey, ?array $configargs = null, ?array $extraattribs = null): OpenSSLCertificateSigningRequest|false {}
+/** @param OpenSSLAsymmetricKey $private_key */
+function openssl_csr_new(array $distinguished_names, &$private_key, ?array $options = null, ?array $extra_options = null): OpenSSLCertificateSigningRequest|false {}
 
-function openssl_csr_get_subject(OpenSSLCertificateSigningRequest|string $csr, bool $use_shortnames = true): array|false {}
+function openssl_csr_get_subject(OpenSSLCertificateSigningRequest|string $request, bool $short_names = true): array|false {}
 
-function openssl_csr_get_public_key(OpenSSLCertificateSigningRequest|string $csr, bool $use_shortnames = true): OpenSSLAsymmetricKey|false {}
+function openssl_csr_get_public_key(OpenSSLCertificateSigningRequest|string $request, bool $short_names = true): OpenSSLAsymmetricKey|false {}
 
-function openssl_pkey_new(?array $configargs = null): OpenSSLAsymmetricKey|false {}
+function openssl_pkey_new(?array $options = null): OpenSSLAsymmetricKey|false {}
 
-/**
- * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $key
- * @param string $out
- */
-function openssl_pkey_export_to_file($key, string $outfilename, ?string $passphrase = null, ?array $configargs = null): bool {}
+/** @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $private_key */
+function openssl_pkey_export_to_file($private_key, string $output_filename, ?string $passphrase = null, ?array $options = null): bool {}
 
 /**
- * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $key
- * @param string $out
+ * @param OpenSSLAsymmetricKey|OpenSSLCertificate|array|string $private_key
+ * @param string $output
  */
-function openssl_pkey_export($key, &$out, ?string $passphrase = null, ?array $configargs = null): bool {}
+function openssl_pkey_export($private_key, &$output, ?string $passphrase = null, ?array $options = null): bool {}
```

Comment:
According to the `php_openssl_pkey_from_zval(zpkey, 0, passphrase, passphrase_len);` invocation, it isn't. But I'll double check it.

Reply:
Yes, even in the `ext/openssl/tests/openssl_pkey_export_basic.phpt` test, a public key is used with `openssl_pkey_get_details()`

Reply:
Ok I see it in the code but it's a bit artificial as it could be possible to extend it and support pubkey using `PEM_write_bio_PUBKEY`. Then the name wouldn't be correct and we couldn't easily change it. As `pkey` is already in the function name, I think it should be fine to rename the argument to `pkey`. It should be done for other cases as well.

Reply:
I think this is the last remaining (important) question. @nikic Do you have any preference? To be honest, I don't like the `pkey` name at all, because it made me google for quite long whether it refers to a private or a public key when I was working on the resource migration. :( So I'd favor a `$key`, `$asymmetric_key` name if we really want to support public keys too.

Reply:
Just `$key` sounds reasonable to me here. But really, no particular preference. I do think we should use $public_key and $private_key in the APIs that only accept one of them.

Reply:
I went with `$key` in case of `openssl_pkey_export()` and `openssl_pkey_export_to_file()` to be a bit more forward-compatible :)